app/battle/battle_manager.py

```diff
@@ -33,6 +33,13 @@ def apply_attributes(self) -> None:
             knight.apply_potion()
 
     def battle(self) -> dict[str, int]:
+        def duel(knight_1: Knight, knight_2: Knight) -> None:
```

Comment:
okay, but not save it in another function, save it in BattleManager class with `@staticmethod`